src/main/java/org/example/servletsHomework/servlet/TextsServlet.java

```diff
@@ -0,0 +1,153 @@
+import org.example.servletsHomework.exception.NotFoundException;
+import org.example.servletsHomework.handler.TextCreateRequest;
+import org.example.servletsHomework.handler.TextCreateResponse;
+import org.example.servletsHomework.model.IdGenerator;
+import org.example.servletsHomework.model.Texts;
+import org.example.servletsHomework.service.TextService;
+import org.example.servletsHomework.storage.TokensAndUserStorage;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class TextsServlet extends BaseServlet {
+
+    private final ObjectMapper objectMapper;
+    private final IdGenerator idGenerator;
+    private final TextService textService;
+    private final TokensAndUserStorage tokensAndUserStorage;
+
+    public TextsServlet(ObjectMapper objectMapper, IdGenerator idGenerator, TextService textService, TokensAndUserStorage tokensAndUserStorage) {
+        this.objectMapper = objectMapper;
+        this.idGenerator = idGenerator;
+        this.textService = textService;
+        this.tokensAndUserStorage = tokensAndUserStorage;
+    }
+
+    private static final String INTERNAL_SERVER_ERROR = "Internal Server Error";
+    private static final String EMPTY_TEXT_ERROR = "Text cannot be empty";
+    private static final String INVALID_REQUEST_PATH = "Invalid request path";
+    private static final String TEXT_NOT_FOUND = "Text not found";
+    private static final String HEADER_NAME_TOKEN = "token";
+
+
+    @Override
+    public void handlePost(HttpServletRequest req, HttpServletResponse resp) {
+
+        resp.setContentType("application/json");
+        String token = req.getHeader(HEADER_NAME_TOKEN);
+
+        try {
+            PrintWriter respWriter = resp.getWriter();
+
+            TextCreateRequest requestBody = objectMapper.readValue(req.getReader(), TextCreateRequest.class);
+            TextCreateResponse responseBody = new TextCreateResponse();
+            String text = requestBody.getText();
+
+            if (text == null || text.trim().isEmpty()) {
+                throw new BadRequestException(EMPTY_TEXT_ERROR);
+            } else {
+                long userId = tokensAndUserStorage.getUserIdByToken(token);
+                long textId = idGenerator.getNextId();
+                textService.addText(userId, new Texts(textId, text));
+
+                responseBody.setTextId(textId);
+                responseBody.setText(text);
+                responseBody.setMessage("Text saved with id: " + textId);
+
+                resp.setStatus(HttpServletResponse.SC_CREATED);
+                respWriter.write(objectMapper.writeValueAsString(responseBody));
+            }
+        } catch (IOException e) {
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
+        }
+
+    }
+
+
+    @Override
+    public void handleGet(HttpServletRequest req, HttpServletResponse resp) {
+
+        String token = req.getHeader(HEADER_NAME_TOKEN);
```

Comment:
you can put something into request's attributes. It would be easier if in servlet I would just do `Long userId = req.getAttribute("userId"); `